internal/ir/xds.go

```diff
@@ -350,7 +350,7 @@ type TLSConfig struct {
 	// SignatureAlgorithms supported by this listener
 	SignatureAlgorithms []string `json:"signatureAlgorithms,omitempty" yaml:"signatureAlgorithms,omitempty"`
 	// ALPNProtocols exposed by this listener
-	ALPNProtocols []string `json:"alpnProtocols,omitempty" yaml:"alpnProtocols,omitempty"`
+	ALPNProtocols *[]string `json:"alpnProtocols,omitempty" yaml:"alpnProtocols,omitempty"`
```

Comment:
is the ptr needed ? the zero value of a slice is `nil`

Reply:
Sure. If we keep omitEmpty, the .out files would omit alpnProtocols both for nil and empty slices, so that can make IR/XDS test reading a bit confusing. If we remove it, tests are a bit inflated with `alpnProtocols: null`. Any preference?

Reply:
ah thanks for explaining why you chose this route :) even if .out files omit alpnProtocols, the test title (-empty-alpn) should hopefully solve the confusion imo

Reply:
ok, my bad, it's not just a readability issue. When .out files are parsed, all the omitted slices are assumed to be nil, while the IR structs that they are are compared to might be an empty slice (e.g. for TLSRoutes). The .out testdata would always be wrongly generated in this case. 
So, options are:
- Don't omitempty
- Customize JSON marshaling with additional tags, methods
- Pointer to slice

I added the first option in a new commit.